vendor/bat-native-ads/src/bat/ads/internal/experiments/experiments.h

```diff
@@ -0,0 +1,28 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BAT_ADS_INTERNAL_EXPERIMENTS_EXPERIMENTS_H_
+#define BAT_ADS_INTERNAL_EXPERIMENTS_EXPERIMENTS_H_
+
+namespace ads {
+
+class AdsImpl;
+
+class Experiments {
+ public:
+  Experiments(
+      AdsImpl* ads);
+
+  ~Experiments();
+
+  void Log();
+
+ private:
+  AdsImpl* ads_;  // NOT OWNED
```

Comment:
not sure I understand why do you need this class and this field. Maybe the whole thing should be just a free function?

Reply:
same as "future-proof" comment above

Reply:
I don't think it's necessary to add this now to make things future-proof. Changing the constructor is trivial and does not require extensive refactoring so imo this kind of stuff should be left out unless needed because it's more confusing to have it and not use it

Reply:
resolved with refactor